workflow/controller/operator.go

```diff
@@ -481,6 +481,13 @@ func (woc *wfOperationCtx) setGlobalParameters(executionParameters wfv1.Argument
 	woc.globalParams[common.GlobalVarWorkflowServiceAccountName] = woc.execWf.Spec.ServiceAccountName
 	woc.globalParams[common.GlobalVarWorkflowUID] = string(woc.wf.ObjectMeta.UID)
 	woc.globalParams[common.GlobalVarWorkflowCreationTimestamp] = woc.wf.ObjectMeta.CreationTimestamp.Format(time.RFC3339)
+	if annotation := woc.wf.ObjectMeta.GetAnnotations(); annotation !=nil {
+		val, ok := annotation[common.AnnotationKeyCronWfScheduledTime]
+		if ok {
+			woc.globalParams[common.GlobalVarWorkflowCronScheduleTime] = val
+		}
+	}
+
```

Comment:
I'm pretty sure you can simplfy this to

```golang
woc.globalParams[common.GlobalVarWorkflowCronScheduleTime] = woc.wf.GetAnnotations()[common.AnnotationKeyCronWfScheduledTime]
```

Reply:
nil check not needed, and empty string would be fine

Reply:
what value is this if we `argo submit --from cronwf/...`?? now presumably?

Reply:
I think as a user, I want to use `submit --from` to run a historical version, so there should be an easy way to do this. Would it be better for it to be changed back to a parameter?

Reply:
Recent refactor will not allow the user to pass the scheduledTime. It will pass `now`. If we rollback to the original parameter implementation, user can send the scheduledTime with `-p`. I am fine to rollback to `workflow.parameters.scheduledTime` . Please agreed one.

Reply:
@simster7 thoughts?

Reply:
> I think as a user, I want to use submit --from to run a historical version, so there should be an easy way to do this. Would it be better for it to be changed back to a parameter?

This is a great use case, I think it should definitely be supported. I think this could still be accomplished without `-p`. A flag specific for this use case (eg `--as-schedule` would be fine.

```
argo submit --from cwf/.. --as-schedule [...]
```

Reply:
I like this idea add a new flag in the command line. This will solve the  --from and keep the `workflow.scheduledTime` @alexec are you ok with this?

Reply:
I agree. Maybe call the new argument —scheduled-time ?